app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,119 @@
+from typing import Any, Optional, Union
+
+from app.data_models import QuestionnaireStore
+from app.questionnaire import Location, QuestionnaireSchema, QuestionSchemaType
+from app.questionnaire.placeholder_renderer import PlaceholderRenderer
+from app.questionnaire.variants import transform_variants
+
+
+class PreviewQuestion:
+    def __init__(
+        self,
+        schema: QuestionnaireSchema,
+        questionnaire_store: QuestionnaireStore,
+        section_id: str,
+        block_id: str,
+        language: str,
+    ):
+        self.schema = schema
+        self.questionnaire_store = questionnaire_store
+        self.current_location = Location(section_id=section_id, block_id=block_id)
+        self.block_id = block_id
+
+        self.placeholder_renderer = PlaceholderRenderer(
+            language=language,
+            answer_store=self.questionnaire_store.answer_store,
+            list_store=self.questionnaire_store.list_store,
+            metadata=self.questionnaire_store.metadata,
+            response_metadata=self.questionnaire_store.response_metadata,
+            schema=self.schema,
+            location=self.current_location,
+            preview_mode=True,
+        )
+
+        self.question = self.rendered_block().get("question")
+        # render_block returns same type as placeholder_renderer.render which is dict[str, Any] hence all type ignores below
+
+        self.title = self.question.get("title")  # type: ignore
+
+        self.answers = self._build_answers()
+        self.descriptions = self._build_descriptions()
+        self.guidance = self._build_question_guidance()
+        self.text_length = self._get_length()
+        self.instruction = self.question.get("instruction")  # type: ignore
+        self.answer_description = self._build_answer_descriptions()
+        self.answer_guidance = self._build_answer_guidance()
+
+    def _build_answers(self) -> list[Optional[str]]:
+        labels: list = []
+        if answers := self.question.get("answers"):  # type: ignore
+            for answer in answers:
+                if options := answer.get("options"):
+                    labels.extend(option["label"] for option in options)
+                if not options:
+                    labels.append(answer["label"])
+
+        return labels
+
+    def _build_answer_descriptions(self) -> Optional[dict]:
+        answers = self.question["answers"]  # type: ignore
+        return next(
+            (
+                answer.get("description")
+                for answer in answers
+                if answer.get("description")
+            ),
+            None,
+        )
```

Comment:
Won't this only return a single description?. What if every answer had a description?.
Each answer need to be built and passed as a list of dict. Each answer object could have things like:

For example a built list may look like:
```py
answers = [
            {
                "label": "some label",
                "description": ["some description"],
                "max_length": 2000
            }
        ]
```

Reply:
Ok so it returns a list of answers now, had to alter it a bit for answer options, instead of label I created a nested `options` now.